Mathlib/Analysis/Calculus/Deriv/Comp.lean

```diff
@@ -164,34 +236,77 @@ theorem HasDerivWithinAt.comp (hh₂ : HasDerivWithinAt h₂ h₂' s' (h x))
   exact hh₂.scomp x hh hst
 #align has_deriv_within_at.comp HasDerivWithinAt.comp
 
-/-- The chain rule. -/
+/-- The chain rule.
+
+Note that the function `h₂` is a function on an algebra. If you are looking for the chain rule
+with `h₂` taking values in a vector space, use `HAsDerivAt.scomp`. -/
 nonrec theorem HasDerivAt.comp (hh₂ : HasDerivAt h₂ h₂' (h x)) (hh : HasDerivAt h h' x) :
     HasDerivAt (h₂ ∘ h) (h₂' * h') x :=
   hh₂.comp x hh hh.continuousAt
 #align has_deriv_at.comp HasDerivAt.comp
 
+/-- The chain rule.
+
+Note that the function `h₂` is a function on an algebra. If you are looking for the chain rule
+with `h₂` taking values in a vector space, use `HAsDerivAt.scomp_of_eq`. -/
+theorem HasDerivAt.comp_of_eq
+    (hh₂ : HasDerivAt h₂ h₂' y) (hh : HasDerivAt h h' x) (hy : y = h x) :
+    HasDerivAt (h₂ ∘ h) (h₂' * h') x := by
+  rw [hy] at hh₂; exact hh₂.comp x hh
+
+theorem HasDerivAtFilter.comp_of_eq (hh₂ : HasDerivAtFilter h₂ h₂' y L')
+    (hh : HasDerivAtFilter h h' x L) (hL : Tendsto h L L') (hy : y = h x) :
+    HasDerivAtFilter (h₂ ∘ h) (h₂' * h') x L := by
+  rw [hy] at hh₂; exact hh₂.comp x hh hL
```

Comment:
Why not put it next to `HasDerivAtFilter.comp`?

Reply:
Thanks for spotting this! Fixed.